TACC/Core-CMS-Custom
# task/WP-404: Add max lengths on certain registration form inputs + bug/WP-406: Handle empty license and/or NAIC codes on DB write

## Overview
Addresses issues which were causing occasional failure to write entities or contacts to the database
…

## Related

- [WP-404](https://jira.tacc.utexas.edu/browse/WP-404)
- [WP-406](https://jira.tacc.utexas.edu/browse/WP-406)
…

## Changes
- Added `maxlength` attr to the inputs corresponding to Entity name, Contact Type, Contact Name, and Contact Email with the specified lengths from the ticket
- Added check for NAIC and license codes at time of db write (entity creation or update) to see if empty string, and write None in the event of zero-length string
…

## Testing

1. Go to https://localhost:8000/register/request-to-submit
2. Fill out the form as normal up to the entities, then confirm:
     - For entity name, no more than 255 characters can be entered on field (can use example below, exactly 255 characters)
        <details><summary>Example string</summary>

           APCD-UTHTX TACCAPCD-UTHTX TACCAPCD-UTHTX TACCAPCD-UTHTX TACCAPCD-UTHTX TACCAPCD-UTHTX TACCAPCD-UTHTX TACCAPCD-UTHTX TACCAPCD-UTHTX TACCAPCD-UTHTX TACCAPCD-UTHTX TACCAPCD-UTHTX TACCAPCD-UTHTX TACCAPCD-UTHTX TACCAPCD-UTHTX TACCAPCD-UTHTX TACCAPCD-UTHTX TACC

        </details> 
     - On at least one entity, do not fill in either license or NAIC code fields (or both)
     - On contacts, that contact type and contact name inputs do not allow more than 100 characters (example below)
        <details><summary>Example string</summary>

           APCD User APCD User APCD User APCD User APCD User APCD User APCD User APCD User APCD User APCD User

        </details> 
     - For contact email, that the email entered cannot be longer than 50 characters (example below)
        <details><summary>Example string</summary>

           thisisalongnameemail@longersecondpartdeflonger.net

        </details> 
3. Repeat step 2 and all sub-steps for the following:
     - https://localhost:8000/administration/list-registration-requests > any record > Actions column > Edit Record
     - https://localhost:8000/register/list-registration-requests > any record > Actions column > Renew Registration

## UI

…
